src/scrot.c

```diff
@@ -606,33 +637,44 @@ static char *imPrintf(char *str, struct tm *tm, char *filenameIM,
             c++;
             switch (*c) {
             case 'a':
-                gethostname(buf, sizeof(buf));
-                strlcat(ret, buf, sizeof(ret));
+                if (hostNameMax == 0) {
+                    /* freebsd and macos doesn't have HOST_NAME_MAX defined.
+                     * instead sysconf is recommended by freebsd */
+                    if ((hostNameMax = sysconf(_SC_HOST_NAME_MAX)) < 0)
+                        errx(EXIT_FAILURE, "couldn't grab _SC_HOST_NAME_MAX");
+                    ++hostNameMax; /* for the nul terminator */
+                }
+
+                streamReserve(&ret, hostNameMax);
+                char *target = ret.buf + ret.off;
+                gethostname(target, hostNameMax);
```

Comment:
One thing I noticed while writing my example was that maybe `{HOST_NAME_MAX}` could change between the `sysconf()` call and the `gethostname()` call. POSIX allows the value to change, but I don't know under what circumstances it may change or if it changes at all on a real OS. Maybe we should `memchr()` for the terminator and add our own if there's none to prevent a possible buffer overread. Let that edge case truncate instead of adding the fancy code to try again. Or ignore it and blame POSIX.